x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -137,6 +139,19 @@ export const createSecurityRuleTypeWrapper: CreateSecurityRuleTypeWrapper =
               ? TIMESTAMP
               : undefined;
 
+          // If we have a timestampOverride, we'll compute a runtime field that emits the override for each document if it exists,
+          // otherwise it emits @timestamp. If we don't have a timestamp override we don't want to pay the cost of using a
+          // runtime field, so we just use @timestamp directly.
+          const { aggregatableTimestampField, timestampRuntimeMappings } =
+            secondaryTimestamp && timestampOverride
+              ? {
+                  aggregatableTimestampField: TIMESTAMP_RUNTIME_FIELD,
+                  timestampRuntimeMappings: buildTimestampRuntimeMapping({
+                    timestampOverride,
+                  }),
+                }
+              : { aggregatableTimestampField: TIMESTAMP, timestampRuntimeMappings: undefined };
```

Comment:
```suggestion
              : { aggregatableTimestampField: primaryTimestamp, timestampRuntimeMappings: undefined };
```
If there's no secondary timestamp and we don't need the runtime field, the primary timestamp could be either the timestamp override if it exists, or `@timestamp`, so we should use `primaryTimestamp` for the aggregatable timestamp rather than `TIMESTAMP`